### PR TITLE
[meetings-sidepanel]: SidePanel in meeting experience csharp sample

### PR DESCRIPTION
**Pull request by Trinetra-MSFT**
## Description
SidePanel sample demonstrates how to use sidepanel tab and update the data at realtime.
This sample uses signalR to update the data at realtime.

Language Used: C#

## Testing
1. Sample have an Agenda tab created which can be editable by organizer and visible to all the participants in meeting
2. Once organizer updates the agenda item in sidepanel tab,it reflect the added item to all the participants

**Steps:**
1. Add SidePanel to meeting chat
2. Non -organizer view
image

3. Organizer View who can add the agenda item
image

4. When click on Publish Agenda, it sent in Chat
image

### CODE REVIEW

File: samples/meetings-sidepanel/csharp/Side Panel/Views/Home/configure.cshtml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+<head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <title>Side Panel Config page</title>
+    <style>
+        body {
+            margin: 0px;
+            padding: 0px;
+            font-family: Segoe UI;
+        }
```
Comment (Wajeed-msft) [inline samples/meetings-sidepanel/csharp/Side Panel/Views/Home/configure.cshtml:20]: Could you please move styles from all the cshtml pages to separate files?

File: samples/meetings-sidepanel/csharp/Side Panel/Model/configure.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
```
Comment (Wajeed-msft) [inline samples/meetings-sidepanel/csharp/Side Panel/Model/configure.cs:4]: Please check and remove if this class is not being used anywhere.

File: samples/meetings-sidepanel/csharp/Side Panel/Model/TaskInfo.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+using System.Threading.Tasks;
+
+namespace SidePanel.Models
+{
+    public class TaskInfo
+    {
+        public string title { get; set; }
+    }
+
+    public class taskList
```
Comment (Wajeed-msft) [inline samples/meetings-sidepanel/csharp/Side Panel/Model/TaskInfo.cs:13]: Please check and remove if this class is not being used anywhere.

File: samples/meetings-sidepanel/csharp/Side Panel/Model/sidePanel.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SidePanel.Models
+{
+    public class sidePanel
+    {
```
Comment (Wajeed-msft) [inline samples/meetings-sidepanel/csharp/Side Panel/Model/sidePanel.cs:9]: Please check and remove if this class is not being used anywhere.

File: samples/meetings-sidepanel/csharp/Side Panel/Manifest/manifest.json [hunk 5]
```diff
@@ -0,0 +1,60 @@
+{
+  "$schema": "https://raw.githubusercontent.com/OfficeDev/microsoft-teams-app-schema/preview/DevPreview/MicrosoftTeams.schema.json",
```
Comment (Wajeed-msft) [inline samples/meetings-sidepanel/csharp/Side Panel/Manifest/manifest.json:2]: Please use the latest 1.9 schema.

File: samples/meetings-sidepanel/csharp/Side Panel/Bots/SidePanelBot.cs [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+            TeamsMeetingParticipant participant = await GetMeetingParticipantAsync(turnContext, meetingId, userId, tenantID);
+            TeamsChannelAccount member = participant.User;
+            MeetingParticipantInfo meetingInfo = participant.Meeting;
+            ConversationAccount conversation = participant.Conversation;
+            var meetingRole = meetingInfo.Role;
+
+            if(meetingRole == "Organizer")
+            {
+                HomeController.meetingOrganizerId = turnContext.Activity.From.AadObjectId;
```
Comment (Wajeed-msft) [inline samples/meetings-sidepanel/csharp/Side Panel/Bots/SidePanelBot.cs:36]: This part if bit confusing. Organizer need to send a message to Bot for this sample to work. Can we instead use Connector client directly to fetch participant details? Similar to this: https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/v-trkum/sidepanel/samples/meetings-sidepanel/csharp/Side%20Panel/Controllers/HomeController.cs#L85

File: samples/meetings-sidepanel/csharp/Side Panel/appsettings.json [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
+  "MicrosoftAppId": "<<Microsoft-App-ID>>",
+  "MicrosoftAppPassword": "<<Microsoft-App-Secret>>",
+  "BaseUrl": "<<Bot-EndPoint-URL>>",
+  "ServiceUrl": "<<Service-URL>>"
```
Comment (Wajeed-msft) [inline samples/meetings-sidepanel/csharp/Side Panel/appsettings.json:5]: This is not required to be part of configuration. When App is installed to the meeting, in conversation update event you can fetch this info.

File: samples/meetings-sidepanel/csharp/Side Panel/Views/Shared/_Layout.cshtml [hunk 8]
```diff
@@ -0,0 +1,20 @@
+﻿<!DOCTYPE html>
+<html lang="en">
+<head>
+    <title>Microsoft Teams Hello World Sample App</title>
+    <link rel="stylesheet" href="~/css/msteams-16.css" />
+    <link rel="stylesheet" href="~/css/Custom.css" />
+    <link rel="stylesheet" href="~/css/Site.css" />
+
+    <script src="https://statics.teams.cdn.office.net/sdk/v1.5.2/js/MicrosoftTeams.min.js" asp-append-version="true"></script>
```
Comment (Wajeed-msft) [inline samples/meetings-sidepanel/csharp/Side Panel/Views/Shared/_Layout.cshtml:9]: Please use latest SDK.

File: samples/meetings-sidepanel/csharp/Side Panel/Views/Shared/_EmbedPage.cshtml [hunk 9]
```diff
@@ -0,0 +1,40 @@
+﻿<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta name="viewport" content="width=device-width" />
+    <title>@ViewBag.Title</title>
+    <script src="https://statics.teams.cdn.office.net/sdk/v1.5.2/js/MicrosoftTeams.min.js" asp-append-version="true"></script>
```
Comment (Wajeed-msft) [inline samples/meetings-sidepanel/csharp/Side Panel/Views/Shared/_EmbedPage.cshtml:6]: Please use latest SDK.

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

File: samples/meetings-sidepanel/csharp/Side Panel/Bots/SidePanelBot.cs (hunk 6 above)
Comment (Trinetra-MSFT) [reply samples/meetings-sidepanel/csharp/Side Panel/Bots/SidePanelBot.cs:36]: We tried to get the participants details but roles is not coming need to check this once

File: samples/meetings-sidepanel/csharp/Side Panel/Model/sidePanel.cs (hunk 4 above)
Comment (Wajeed-msft) [reply samples/meetings-sidepanel/csharp/Side Panel/Model/sidePanel.cs:9]: I see that this is not used, could you please help to remove this?
image

File: samples/meetings-sidepanel/csharp/Side Panel/Model/configure.cs (hunk 2 above)
Comment (Wajeed-msft) [reply samples/meetings-sidepanel/csharp/Side Panel/Model/configure.cs:4]: This is not used. 
image

File: samples/meetings-sidepanel/csharp/Side Panel/Bots/SidePanelBot.cs (hunk 6 above)
Comment (Wajeed-msft) [reply samples/meetings-sidepanel/csharp/Side Panel/Bots/SidePanelBot.cs:36]: Is this resolved?

File: samples/meetings-sidepanel/csharp/README.md [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+6. Update the manifest.json file with ```Microsoft-App-ID``` and ```BaseUrl``` value.
+7. Run your app, either from Visual Studio with ```F5``` or using ```dotnet run``` in the appropriate folder.
+8. [Install the App in Teams Meeting](https://docs.microsoft.com/en-us/microsoftteams/platform/apps-in-teams-meetings/teams-apps-in-meetings?view=msteams-client-js-latest#meeting-lifecycle-scenarios)
+
+## Interacting with the app in Teams Meeting
+Interact with SidePanel by clicking on the App icon present on the top menu beside the "more actions" during a meeting.
+1. Organizer need to ping the bot by mentioning ```@``` in the meeting chat in order to gain the Organizer privilage of the app to Add and Publish the Agenda points.
+2. Once the app is clicked, sidepanel appears with the default agenda list. Only organizer gets the feasibility to add new agenda points to the list using "Add New Agenda Item" button.
+3. On click of "Add" button, agenda point will be added to the agenda list.
+4. On click of "Publish Agenda", the agenda list will be sent to the meeting chat.
```
Comment (Wajeed-msft) [inline samples/meetings-sidepanel/csharp/README.md:50]: Add images reference which are added in description to readme file.  Copied references here: 

image

image

image

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

**Review by Wajeed-msft — APPROVED** [review]

**Review by asnow003 — APPROVED** [review]

**Pull request merged**